addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,155 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OneWireGPIOHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OneWireGPIOHandler.class);
+
+    private String gpioBusFile;
+    private Integer refreshTime;
+
+    public OneWireGPIOHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(TEMPERATURE)) {
+            if (command instanceof RefreshType) {
+                publishSensorValue(channelUID);
+            } else {
+                logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        updateStatus(ThingStatus.UNKNOWN);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        Configuration configuration = getConfig();
+        gpioBusFile = (String) configuration.get(GPIO_BUS_FILE);
+        if (gpioBusFile == null) {
+            logger.warn("GPIO_BUS_FILE not set. Please check configuration, and set proper path to w1_slave file.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid path to w1_slave file");
+            return;
+        }
+
+        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
+        if (refreshTime.intValue() <= 0) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+                    DEFAULT_REFRESH_TIME);
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    publishSensorValue(channel.getUID());
+                }
+            }
+        };
+
+        scheduler.scheduleWithFixedDelay(refresher, 0, refreshTime.intValue(), TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh every {} seconds", refreshTime.intValue());
+    }
+
+    private void publishSensorValue(ChannelUID channelUID) {
+        String channelID = channelUID.getId();
+        State state = null;
+        switch (channelID) {
+            case TEMPERATURE:
+                BigDecimal temp = readSensorTemperature(gpioBusFile);
```

Comment:
It may be a good idea to create the actual state down in the `readSensorTemperature` method (maybe in a separate private method) and check for `null` state here in case you could not read the temperature. See my comments below.